### DISCUSSION TITLE
Ketcher 2.3.0 (July 22, 2021)

### DISCUSSION THREAD

**Discussion by AndreiMazol** [epam/ketcher#687] [General]
**Summary**

We are glad to announce that a new stable version of Ketcher [2.3.0](https://github.com/epam/ketcher/milestone/13?closed=1) is released. 
The main features implemented in scope of this release are support of enhanced stereochemistry, ability to save structure as an image file, text tool, and support of multiple reaction arrows.

**New features and improvements** 

* Stereo Flags shouldn't appear for structures without correct stereocenters #418

* It's impossible to open Ketcher with already entered molecule #601

* Add a warning when saving a reaction equilibrium arrow in a format that does not support it #543

* Add possibility to replace api path while App is building by passing additional argument #507

* Add multiple arrow support for the same canvas #145

* SETTINGS for the STEREO FLAGS and STEREO MARKS #60

* SAVE and OPEN STRUCTURE with STEREO MARKS #64

* Set PUBLIC_URL as './' by default #508

* Add possibility to add plain text on canvas #144

* Colors for the STEREO FLAGS and STEREO MARKS #62

* Implement "Create Stereo Marks" window #417

* Add support for hydrogen bonds #72

* Representation of Stereo Flags #57

* Stereocenter should disappear after replacing the stereobond with the dative bond or the double cis/trans bond #554

* Implement "Enhanced Stereochemistry" window #527

* Chiral flag should be removed when all Up/Down stereobonds are removed #61

* User can change the text of the four Stereo flags #518

* Rename the "Do not show the Stereo flags" setting and change the default value #556

* Rendering only feature #107

* Add possibility to save structure as image #437

* Unclear/inconsistent handling of absolute/relative stereochemistry #290

* Label Display At Stereogenic Centers for the STEREO FLAGS and STEREO MARKS #63

* Extend Text tool by bold, italic, subscript and superscript support #339

* Move dative/hydrogen bond tools to a separate block in the 'Bond' tool menu #538

* Add the ability to save information about the coordinates of the Stereo flags in * .ket #559

* Update position of stereo flag while structure is moved #167

**Bugfixes**

* Adding a stereo flag when opening the structure #632

* Incorrect template addition #676

* The descriptor overlaps the structure after Layout #219

* Layout action does not work on the distorted S-group members #671

* No alert error message when trying to open incorrect InChi string #657

* "Undo" after template merging does not work correctly #670

* Stereo-label doesn't update after stereoBond flipping #661

* Left toolbar is not visible in Firefox #612

* Redo does not work for simple object "Shape line" after moving its dots #635

* IE: Save as *.svg file does not work #627

* Atom is not underlined in red after adding to the template #636

* Application crashes after saving a template #652

* The incorrect structure when marks "Load as fragment" #633

* IE: The structure does not open #624

* The application doesn't work after "Undo" #615

* When marks "Load as fragment and copy to the Clipboard" structure don't save to clipboard #634

* Modal windows don't close when pressing the esc button #626

* The structure with R-Group is not visible #621

* Multiple arrows are saved as a reaction in Daylight Smiles format #590

* Ketcher build does not work on Linux #505

* Rotation angle doesn't display after I change its value #380

* 'Calculated values' - The data in 'Elemental analysis' field is displayed incorrectly #445

* Calculated values: old values blink before displaying new once #389

* 'Add text' - The initially created text object remains selected after adding the second one #464

* 'Add text' - It's not obvious how to reduce the "Text editor" window when a scroll is present in it #473

* 'Add text'- Impossible to edit a single letter text object #470

* 'Add text' - Double scroll appears when a long text is entered into the "Text editor" window #467

* 'Save Structure' and 'Structure Check' windows are enlarged when trying to save/check a big molecule #454

* It is possible to save a structure in incorrect format when network connection is slow #456

* The error occurs when trying to undo the delete action for a structure with stereo bonds #485

* Blank page appears after trying to save a reaction with an R-Group member #488

* A part of S-group brackets is enlarged after Cut - Undo actions #448

* 'Add text'- 'Cut'/'Copy' buttons become enabled when a text object is added to the canvas #472

* Incorrect rendering of Stereo Flags in reactions #58

* Console error after delete - undo/redo actions with a structure with stereobonds #573

* "Help" module doesn't display correctly in IE11 #572

* Incorrect stereo flag calculation #579

* Browser crash when copy and paste a sgroup structure #592

* Stereo labels after pasting display incorrectly #194

* The Stereo labels and Attached Data keep their positions after the Flip action #201

* Stereo Flag is not highlighted #604

* Vertical/Horizontal flip hot keys don't work correctly #524

* Error message when copying structure #606

* Blank page and a console error appear while saving #514

* Simple objects - An error in console after the 'Undo' action #511

* 'Add text'- 'Cut'/'Copy' buttons are disabled when a text object is added to the canvas #522

* Error message when trying to open *.ket files with stereochemistry features #532

* 3D Viewer - The light mode doesn't work #531

* Stereo flags are confused with stereo marks in *.ket files #544

* Unable to paste the text objects after cut/copy #551

* The periodic table opens when a text object is selected with the Lasso/Rectangle/Fragment selection tools #549

* Stereo label (E) saves in .smi file #208

* Wrong modal window, a blank page and an error in console after trying to add the Attachment points/R-sites to a structure #499

<hr /><em>This discussion was created from the release <a href='https://github.com/epam/ketcher/releases/tag/v2.3.0'>Ketcher 2.3.0 (July 22, 2021)</a>.</em>

**Comment by robtree** [comment]
Congratulations on the latest release!

Do you recommend specific minimum IFRAME dimensions to avoid scrollbars appearing on the canvas and toolbars in 2.3?

https://lifescience.opensource.epam.com/ketcher/demo/ketcher_2_3_remote.html

![image](https://user-images.githubusercontent.com/3150192/127523153-a23d2497-03ed-4b79-aaae-833aa13689a9.png)

**Reply by AndreiMazol** [reply]
Hi @robtree,

minimum height of IFrame should be 700 px to hide all visible scrolls. The reason of showing them is the fact that panels have been extended by new tools (Enhanced Stereochemistry button added to right panel  and Text tool added to Left panel). As a workaround you can override behavior by your own styles but please be aware that Ketcher team is working on redesign of our peorduct so  I guess new user-friendly approach to show toolbars will be provided in the nearest releases.  

Best regards,
Andrei
